ra-optimisation/search-tree-rao/src/test/java/com/powsybl/openrao/searchtreerao/castor/algorithm/CastorFullOptimizationTest.java

```diff
@@ -551,4 +551,15 @@ void testRaoWithEmptyCrac() throws IOException {
         assertNotNull(raoResult);
         assertEquals(-Double.MAX_VALUE, raoResult.getCost(null));
     }
+
+    @Test
+    void checkWithHvdc() throws IOException {
+        // same test as US 15.17.8
+        setup("TestCase16NodesWithHvdc_AC_emulation.xiidm", "jsonCrac_ep15us12-5case8.json");
+        RaoParameters raoParameters = JsonRaoParameters.read(getClass().getResourceAsStream("/parameters/RaoParameters_DC.json"));
+
+        RaoResult raoResult = new CastorFullOptimization(raoInput, raoParameters, null).run().join();
```

Comment:
could you add this kind of checks raoResult.getActivatedRangeActionsDuringState(crac.getState("co1_be1_fr5", crac.getInstant(InstantKind.CURATIVE))) ? Or do you thinks it's too specific for this function